### ISSUE TITLE
Support AWS EC2 instances without public IP

### ISSUE THREAD

**Issue by csantanapr** [loft-sh/devpod-provider-aws#22]
**Is your feature request related to a problem?**  
Some users can't have the ec2 instance on public subnet, they need to be in private subnet

**Which solution do you suggest?**  
Devpod should be able to leverage the new recent feature ec2 connect https://aws.amazon.com/about-aws/whats-new/2023/06/amazon-ec2-instance-connect-ssh-rdp-public-ip-address/ 

**Which alternative solutions exist?**  
don't know

**Additional context**

**Comment by csantanapr** [comment]
Any updates on this

**Comment by csantanapr** [comment]
Any updates on this?

**Comment by pascalbreuninger** [comment]
@csantanapr we've release a new alpha version to test this, you'd need to
1. run `devpod provider add aws@v0.0.10-alpha.0` 
2. set `devpod provider set-options aws -o AWS_USE_INSTANCE_CONNECT_ENDPOINT` to use the default instance connect endpoint for your subnet or alternatively specify which one you want with the `AWS_INSTANCE_CONNECT_ENDPOINT_ID` option
3. create a new workspace

iirc the maximum duration for tunnels through instance connect endpoints is 1 hour, if you're IDE doesn't do automatic reconnections you'll need to manually ssh into the workspace again

**Comment by shanman190** [comment]
EC2 Instance Connect will be a welcomed feature for those users that don't have a Direct Connect or VPN to their VPCs.

Just to note for others that arrive here as well, with direct connect or VPN, I had added direct private ip support back with https://github.com/loft-sh/devpod-provider-aws/pull/10.

**Comment by csantanapr** [comment]
Nice @pascalbreuninger Nice 👍 Thank you !

**Comment by csantanapr** [comment]
@shanman190 your saying you fix in #10 the ability to use VPN and use the private IP of the instance (ie 10.0.1.3) ?

**Comment by shanman190** [comment]
@csantanapr, yes, if you happen to have a private connection to the VPC such as a VPN, then that PR enabled you to access the instance via its private IP.

**Comment by pascalbreuninger** [comment]
@csantanapr That's currently not easily doable from the UI for alpha and beta versions. You can always quick-install the CLI from `Settings > Add CLI to Path`, though
